test/stringify.js

```
@@ -502,4 +502,25 @@ test('stringify()', function (t) {
 
         st.end();
     });
+
+    t.test('RFC 1738 spaces serialization', function (st) {
+        st.equal(qs.stringify({ a: 'b c' }, { format: 'RFC1738' }), 'a=b+c');
+        st.end();
+    });
+
+    t.test('RFC 3986 spaces serialization', function (st) {
+        st.equal(qs.stringify({ a: 'b c' }, { format: 'RFC3986' }), 'a=b%20c');
+        st.end();
+    });
+
+    t.test('Backward compatibility to RFC 3986 and edge cases', function (st) {
+        st.equal(qs.stringify({ a: 'b c' }, { format: 'UFO1234' }), 'a=b%20c');
+        st.equal(qs.stringify({ a: 'b c' }, { format: false }), 'a=b%20c');
```

I'm wondering if it would be better to throw immediately if the provided format is invalid.